feat(CNF-18687): Implement FBC image resolvability pipeline

fontivan opened the pull request:
Adds a test scenario to ensure all images in a File-Based
Catalog (FBC) are resolvable. A key feature is IDMS support,
allowing validation of pre-release images.

This also adds the following reusable tasks:
- download-file-from-git-repo: Downloads a single file from a
  remote git repository.
- validate-fbc-images-resolvable: Wraps the validation script
  for use in other pipelines.

fontivan committed:
feat(CNF-18687): implement fbc image resolvability pipeline

Adds a test scenario to ensure all images in a File-Based
Catalog (FBC) are resolvable. A key feature is IDMS support,
allowing validation of pre-release images.

This also adds the following reusable tasks:
- download-file-from-git-repo: Downloads a single file from a
  remote git repository.
- validate-fbc-images-resolvable: Wraps the validation script
  for use in other pipelines.

Signed-off-by: Steven Skeard <sskeard@redhat.com>

fontivan commented:
/verify-owners

fontivan commented:
/ok-to-test

fontivan commented:
/verify-owners

fontivan commented:
/verify-owners

rauhersu commented:
/approve

johnbieren commented on tasks/download-file-from-git-repo/OWNERS:
Can you update this?

fontivan replied:
Not sure how I managed to create it and forgot to put the content in, but fixed!

johnbieren commented:
/approve

fontivan committed:
chore(CNF-18687): fix yamllint and shellcheck warnings

Resolves linter warnings in the pipelines and
tasks added in the previous commit

Signed-off-by: Steven Skeard <sskeard@redhat.com>

fontivan committed:
feat(CNF-18687): add test coverage for 2 tasks

The new tasks added in the previous commits now have test coverage:
- download-file-from-git-repo
- validate-fbc-images-resolvable

Signed-off-by: Steven Skeard <sskeard@redhat.com>

johnbieren commented:
@Paul123111 what is `ci/prow/sample-test`?

fontivan commented:
/hold

fontivan commented on pipelines/validate-fbc-images-resolvable/README.md:
````diff
@@ -0,0 +1,82 @@
+        value: "development"
+      - name: pathInRepo
+        value: "pipelines/validate-fbc-images-resolvable/validate-fbc-images-resolvable.yaml"
+```
````
@Paul123111 I see the incoming readme script updates here: https://github.com/konflux-ci/community-catalog/pull/22

is there any way to add extra details like I've done in my current readme?

johnbieren replied:
This is a good point. @Paul123111 maybe in this repo we can offer the auto readme tool but not force it? So run the check, but don't make it mandatory?

Paul123111 replied:
Right now, there's no way to add extra details in the README since the check makes sure both the README and `yaml` file have the same description.

I think `Check README.md files` should be optional in this repo, but I could also add a way to only update the table and check the description and parameter table separately in PRs, since I think the table check would be useful to a lot of tasks and pipelines.

johnbieren replied:
I think we just shouldn't have the github check for it in this repo, but we can still have it described in the docs in case users want it

fontivan replied:
Once the automation is configured I can come back to see if any README changes are required here

fontivan commented on pipelines/validate-fbc-images-resolvable/validate-fbc-images-resolvable.yaml:
```diff
@@ -0,0 +1,134 @@
+      taskRef:
+        resolver: "git"
+        params:
+          # TODO: Before merging, change the url and revision back to the upstream repo and branch
```
@johnbieren @Paul123111 is there any way to have this resolve to itself automatically or do I just need to update it manually before merging?

johnbieren replied:
AFAIK nothing is going to automatically update this, but I can't think of anything preventing you from updating it to this repo's url

fontivan replied:
I had to use my git details here initially so it would actually work since I was using it already, but I do need to update it to this repo and branch before merging so I'll do it manually

johnbieren replied:
makes sense. yeah, you can do it here or in a followup pr (which should be self service)

Paul123111 commented:
> @Paul123111 what is ci/prow/sample-test?

Onboarding this repo to [openshift/release](https://github.com/openshift/release) to use prow/tide required adding at least one test. The other release repos have an e2e test set up in [openshift/release](https://github.com/openshift/release), but there isn't an e2e test for community-catalog, so I added a test that didn't do anything to bypass this.

I'd also like that test gone, but I think I would need an actual test to replace it with.

You can also check the PR the test was added to, there's a few comments relating to that test: https://github.com/openshift/release/pull/65714

fontivan commented:
/unhold

fontivan commented:
@johnbieren despite the `/approve` the bot does't seem to want to merge this, though if I click through to the Tide check it says ready to merge (https://prow.ci.openshift.org/pr?query=is%3Apr+repo%3Akonflux-ci%2Fcommunity-catalog+author%3Afontivan+head%3Asskeard%2Fvalidate-fbc-images-resolvable-pipeline)
```
tide Waiting for status to be reported — Not mergeable. PullRequest is missing sufficient approving GitHub review(s)
```

johnbieren commented:
> @johnbieren despite the `/approve` the bot does't seem to want to merge this, though if I click through to the Tide check it says ready to merge (https://prow.ci.openshift.org/pr?query=is%3Apr+repo%3Akonflux-ci%2Fcommunity-catalog+author%3Afontivan+head%3Asskeard%2Fvalidate-fbc-images-resolvable-pipeline)
> 
> ```
> tide Waiting for status to be reported — Not mergeable. PullRequest is missing sufficient approving GitHub review(s)
> ```

@Paul123111 can you look into this? This shouldn't be required. In the meantime, I'll approve and see if it lets this be merged

johnbieren commented:
> > @johnbieren despite the `/approve` the bot does't seem to want to merge this, though if I click through to the Tide check it says ready to merge (https://prow.ci.openshift.org/pr?query=is%3Apr+repo%3Akonflux-ci%2Fcommunity-catalog+author%3Afontivan+head%3Asskeard%2Fvalidate-fbc-images-resolvable-pipeline)
> > ```
> > tide Waiting for status to be reported — Not mergeable. PullRequest is missing sufficient approving GitHub review(s)
> > ```
> 
> @Paul123111 can you look into this? This shouldn't be required. In the meantime, I'll approve and see if it lets this be merged

I just turned off the "require approvals" setting in the github settings page for the development branch. Maybe that was it, but not sure since it was tide reporting that

Paul123111 commented:
I set that to prevent self approvals from merging in tide - /approve from someone who is not the PR author counts as a sufficient approving GitHub review.

johnbieren commented:
> I set that to prevent self approvals from merging in tide

Well, I added the approved label and I am not the author so something is definitely not working right

Paul123111 commented:
I meant anyone in `konflux-ci` except the PR author

Paul123111 commented:
I'll try to fix any problems that's causing for approvals, though.

Paul123111 commented:
~~I'm wondering if it's not merging because it wants this PR to be rebased, but is incorrectly stating that it doesn't have enough approving reviews. I think the approvals a few days ago were discarded due to new changes in the PR.~~

Nevermind, it looks like it merged after I changed the branch protection rules

Paul123111 commented:
This is closed, but I've realised that checking GitHub's reviews will not work if you don't have write access to the repo, so there was no sufficient review, which is one of the reasons why this wasn't merging. Why it was also ignoring Johnny, I don't know, but I'll fix this as soon as possible.
